Cython/Debugger/libcython.py

```
@@ -165,6 +220,63 @@ def __init__(self,
 
 # General purpose classes
 
+# fail fast if the API changes
+frame_repr_whitelist = set(f.__qualname__ for f in [
+    gdb.Frame.is_valid,
+    gdb.Frame.name,
+    gdb.Frame.architecture,
+    gdb.Frame.type,
+    gdb.Frame.pc,
+    gdb.Frame.block,
+    gdb.Frame.function,
+    gdb.Frame.older,
+    gdb.Frame.newer,
+    gdb.Frame.find_sal,
+    gdb.Frame.select,
+    gdb.Frame.static_link,
```

`gdb.Frame.static_link` was added in GDB 14.1, which is not yet widely available in package managers. This line makes cygdb incompatible with early versions.

Frankly, I don't know why and how this PR passed CI tests, which seem to use GDB 12.

---

Just to double check, is this the only line that needs to be removed to fix the build with GDB 12? I spent a bit trying to set up a docker container with a GDB 12.1 manual install but it was taking more troubleshooting than I wanted.

---

I am unable to check this right now, also troubleshooting. But generally speaking, do you have to evaluate qualnames on import? If you don't, then hard-coding qualnames should work. And if you do, than you could iterate through `dir(gdb.Frame)` or `gdb.Frame.__dict__` and get the methods. This way you could keep `gdb.Frame.static_link` for newer GDB versions, while maintaining compatibility with older versions.

Also, I do not understand the `# fail fast if the API changes` part. I might be missing some important point, but if the "fail fast" is necessary, could it be moved from runtime to unit tests?

---

Iterating over `dir(gdb.Frame)` was how this PR worked initially, but it got changed as a result of [this comment](https://github.com/cython/cython/pull/5958#discussion_r1487460084). I don't remember why I thought it was important to fail fast for API changes. Hard-coding the qualnames is probably fine. I'll try to get my development environment running again.